### PR TITLE
Monitor contention in tokio threadpool

### PR DESCRIPTION
**Pull request by leoyvens**
A simple mechanism that periodically checks the threadpool responsiveness. Part of the investigation for #856.

### CODE REVIEW

File: node/src/main.rs [hunk 1]
```diff
@@ -64,7 +63,6 @@ lazy_static! {
 git_testament!(TESTAMENT);
 
 fn main() {
-    println!("REORG_THRESHOLD = {:#?}", *REORG_THRESHOLD);
```
Comment (olibearo) [inline node/src/main.rs]: 👍

File: node/src/main.rs [hunk 2]
```diff
@@ -574,6 +572,29 @@ fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
...
+    // task that simply responds to "ping" requests. Then spawn a separate
+    // thread to periodically ping it and check responsiveness.
+    let (ping_send, ping_receive) = mpsc::channel::<std::sync::mpsc::Sender<()>>(1);
+    tokio::spawn(
+        ping_receive
+            .for_each(move |pong_send| pong_send.clone().send(()).map(|_| ()).map_err(|_| ())),
+    );
+    let contention_logger = logger.clone();
+    std::thread::spawn(move || loop {
+        std::thread::sleep(Duration::from_secs(10));
```
Comment (olibearo) [inline node/src/main.rs]: My gut is telling me that 10 seconds is too sparse, contention could come and go during that window. 

This seems cheap enough to run every 10ms - 100ms. Is that the case?

File: node/src/main.rs [hunk 3]
```diff
@@ -574,6 +572,29 @@ fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
...
+        let (pong_send, pong_receive) = std::sync::mpsc::channel();
+        ping_send.clone().send(pong_send).wait().unwrap();
+        let mut timeout = Duration::from_millis(1);
+        while pong_receive.recv_timeout(timeout).is_err() {
+            warn!(contention_logger, "Possible contention in tokio threadpool";
```
Comment (olibearo) [inline node/src/main.rs]: If add a `LogCode` here, that would help in making a Grafana panel + alert for this

File: node/src/main.rs (hunk 2 above)
Comment (leodasvacas) [reply node/src/main.rs]: Agreed, I'll lower to 100ms.

File: node/src/main.rs (hunk 3 above)
Comment (leodasvacas) [reply node/src/main.rs]: I'll add a code, but we should probably alert on any `warn` level log.

**Review by ghost — APPROVED** [review]

**Pull request merged**